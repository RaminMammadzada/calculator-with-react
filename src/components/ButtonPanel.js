import React, { useState } from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = props => {
  const initialState = {
    buttonGroups: [
      { buttons: ['AC', '+/-', '%', '/'] },
      { buttons: ['7', '8', '9', 'x'] },
      { buttons: ['4', '5', '6', '-'] },
      { buttons: ['1', '2', '3', '+'] },
      { buttons: ['0', '.', '='] },
    ],
  };

  const [state] = useState(initialState);

  let buttonsView = [];

  const { buttonGroups } = state;
  const { clickHandler } = props;

  buttonsView = (
    <div className="button-panel">
      {buttonGroups.map(group => (
        <div className="button-list" key={`${group.buttons.join('-')}`}>
          {group.buttons.map(button => {
            let btnStyle = 'normal-button';

            if (['/', 'x', '-', '+', '='].includes(button)) {
              btnStyle = 'normal-button operation-button';
            } else if (button === '0') {
              btnStyle = 'zero-style normal-button';
            } else {
              btnStyle = 'normal-button';
            }

            return (
              <Button
                key={`btn-${button}`}
                buttonName={button}
                handleClick={button => clickHandler(button)}
                styling={btnStyle}
              />
            );
          })}
        </div>
      ))}
    </div>
  );

  return (
    <div>
      {buttonsView}
    </div>
  );
};

ButtonPanel.defaultProps = {
  clickHandler: null,
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func,
};

export default ButtonPanel;
