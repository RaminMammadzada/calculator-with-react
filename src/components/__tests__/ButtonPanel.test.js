import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import ButtonPanel from '../ButtonPanel';

const handleClick = () => ({ total: '0', next: '0', operation: 'x' });

describe('Button Panel', () => {
  it('should render the Button panel component', () => {
    const tree = ReactTestRenderer.create(<ButtonPanel clickHandler={handleClick} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should create the Button Panel component', () => {
    render(<ButtonPanel clickHandler={handleClick} />);
    const element = screen.getByText('0');
    expect(element).toBeInTheDocument();
  });
});
