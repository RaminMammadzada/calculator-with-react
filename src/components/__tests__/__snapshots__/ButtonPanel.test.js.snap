// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Panel should render the Button panel component 1`] = `
<div>
  <div
    className="button-panel"
  >
    <div
      className="button-list"
    >
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="normal-button operation-button"
        onClick={[Function]}
        type="button"
      >
        /
      </button>
    </div>
    <div
      className="button-list"
    >
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="normal-button operation-button"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="button-list"
    >
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="normal-button operation-button"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="button-list"
    >
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="normal-button operation-button"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="button-list"
    >
      <button
        className="zero-style normal-button"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="normal-button"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="normal-button operation-button"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
